### ISSUE TITLE
resource

### ISSUE THREAD

**Issue by poetair** [Yard1/hpbandster-sklearn#4]
when resource is 'n_samples', the result may be not right. For the final performance sorting includes the results of small sample size.and In BaseSuccessiveHalving of sklearn, only the last iteration result is used when resource is 'n_samples'.
https://github.com/scikit-learn/scikit-learn/blob/main/sklearn/model_selection/_search_successive_halving.py#L208

**Comment by Poetair** [comment]
In BaseSuccessiveHalving of sklearn, only the last iteration result is used

**Comment by Yard1** [comment]
Yeah, this indeed can be an issue. Would you be interested in opening a PR for this?

**Comment by poetair** [comment]
Yeah, I'll have a try with minimal code changes

**Comment by poetair** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    
        I have opened a PR just now,   so you can check and merge them at your convenience and I look forward to seeing the new version released. 
    
    
        
        

        
    

    On 10/13/2022 00:32，Antoni ***@***.***> wrote： 



Yeah, this indeed can be an issue. Would you be interested in opening a PR for this?

**Comment by poetair** [comment]
font{
        line-height: 1.6;
    }
    ul,ol{
        padding-left: 20px;
        list-style-position: inside;
    }


    
    Have you received my email? I have opened a PR
    
        
        

        
    

    On 10/13/2022 00:32，Antoni ***@***.***> wrote： 



Yeah, this indeed can be an issue. Would you be interested in opening a PR for this?

**Comment by Yard1** [comment]
Yes, I did - I'll merge it tomorrow, thank you a lot!
